securedrop/source_app/main.py

```diff
@@ -126,14 +133,15 @@ def submit():
                     journalist_filename,
                     msg))
         if fh:
-            g.source.interaction_count += 1
-            fnames.append(
-                store.save_file_submission(
-                    g.filesystem_id,
-                    g.source.interaction_count,
-                    journalist_filename,
-                    fh.filename,
-                    fh.stream))
+            for file_submitted in fh:
+                g.source.interaction_count += 1
+                fnames.append(
+                    store.save_file_submission(
+                        g.filesystem_id,
+                        g.source.interaction_count,
+                        journalist_filename,
+                        file_submitted.filename,
+                        file_submitted.stream))
```

Comment:
We should commit after each add so that if the connection breaks, we will have saved some of the files. We should also flash a message "file `foo.txt`" saved per DB commit so the user only has to reupload a portion of the files after a connection breaks.

Reply:
Normally I wouldn't recommend this but since this is over a THS, I think this is advisable.